samza-core/src/main/java/org/apache/samza/clustermanager/container/placement/ContainerPlacementMessageObjectMapper.java

```diff
@@ -0,0 +1,113 @@
+        throws IOException {
+      Map<String, Object> containerPlacementMessageMap = new HashMap<String, Object>();
+      if (value instanceof ContainerPlacementRequestMessage) {
+        containerPlacementMessageMap.put("subType", ContainerPlacementRequestMessage.class.getName());
+      } else if (value instanceof ContainerPlacementResponseMessage) {
+        containerPlacementMessageMap.put("subType", ContainerPlacementResponseMessage.class.getName());
+        containerPlacementMessageMap.put("responseMessage",
+            ((ContainerPlacementResponseMessage) value).getResponseMessage());
+      }
+      containerPlacementMessageMap.put("uuid", value.getUuid().toString());
+      containerPlacementMessageMap.put("applicationId", value.getApplicationId());
+      containerPlacementMessageMap.put("processorId", value.getProcessorId());
+      containerPlacementMessageMap.put("destinationHost", value.getDestinationHost());
+      containerPlacementMessageMap.put("requestExpiry", value.getRequestExpiry());
+      containerPlacementMessageMap.put("statusCode", value.getStatusCode().name());
+      jsonGenerator.writeObject(containerPlacementMessageMap);
+    }
+  }
+
+  static class ContainerPlacementMessageDeserializer extends JsonDeserializer<ContainerPlacementMessage> {
+    @Override
+    public ContainerPlacementMessage deserialize(JsonParser jsonParser, DeserializationContext context)
+        throws IOException {
+      ObjectCodec oc = jsonParser.getCodec();
+      JsonNode node = oc.readTree(jsonParser);
+      String subType = node.get("subType").getTextValue();
+      String applicationId = node.get("applicationId").getTextValue();
+      String processorId = node.get("processorId").getTextValue();
+      String destinationHost = node.get("destinationHost").getTextValue();
+      Long requestExpiry = node.get("requestExpiry").isNull() ? null : node.get("requestExpiry").getLongValue();
+      ContainerPlacementMessage.StatusCode statusCode = null;
+      UUID uuid = UUID.fromString(node.get("uuid").getTextValue());
+      for (ContainerPlacementMessage.StatusCode code : ContainerPlacementMessage.StatusCode.values()) {
+        if (code.name().equals(node.get("statusCode").getTextValue())) {
+          statusCode = code;
+        }
+      }
+      ContainerPlacementMessage message = null;
+      if (subType.equals(ContainerPlacementRequestMessage.class.getName())) {
+        message = new ContainerPlacementRequestMessage(uuid, applicationId, processorId, destinationHost, requestExpiry);
+      } else if (subType.equals(ContainerPlacementResponseMessage.class.getName())) {
+        String responseMessage = node.get("responseMessage").getTextValue();
+        message = new ContainerPlacementResponseMessage(uuid, applicationId, processorId, destinationHost, requestExpiry,
+            statusCode, responseMessage);
+      }
+      return message;
```

Comment:
Could we add this to SamzaObjectMapper and use SamzaObjectMapper instead?

Reply:
Yup SamzaObjectMapper is super bloated. IMO, decoupling is good, especially in this specific domain.

Reply:
The reason its bloated is because the Mixins and custom serializers and deserializers have been defined in the same file. 
You can define yours in separate files but link them in SamzaObjectMapper since this is a part of Samza.

Reply:
"You can define yours in separate files but link them in SamzaObjectMapper since this is a part of Samza."

SamzaObjectMapper class is only used for JobModel serde, I think maybe that is the reason @dnishimura also did not link Startpoint object mappers with it.